src/edgepi/pwm/edgepi_pwm.py

```diff
@@ -41,104 +39,217 @@ def __check_range(self, target, range_min, range_max) -> bool:
         """
         Disable pwm output
+        Args:
+            pwm_num (PWMPins): target pwm device
+        Returns:
+            N/A
         """
-        self.pwm.disable_pwm()
-        self.gpio.set_pin_state(self.pwm_num.value)
+        if pwm_num is None:
+            raise ValueError(f"disable: PWM number is missing {pwm_num}")
+        if pwm_num == PWMPins.PWM1:
+            self.__pwm_1.disable_pwm()
+        else:
+            self.__pwm_2.disable_pwm()
+        self.gpio.set_pin_state(pwm_num.value)
 
-    def get_enabled(self):
+    def get_enabled(self, pwm_num: PWMPins):
         """
         Get enabled state
         Args:
             N/A
         Returns:
             enabled (bool): True enabled, False Disabled
         """
-        return self.pwm.get_enabled_pwm()
+        if pwm_num is None:
+            raise ValueError(f"get_enabled: PWM number is missing {pwm_num}")
+        return self.__pwm_1.get_enabled_pwm() if pwm_num == PWMPins.PWM1 else\
+               self.__pwm_2.get_enabled_pwm()
 
-    def close(self):
+    def close(self, pwm_num: PWMPins):
         """
         Close PWM connection
         Args:
-            N/A
+            pwm_num (PWMPins): target pwm device
         Returns:
             N/A
         """
-        self.pwm.close_pwm()
+        if pwm_num is None:
+            raise ValueError(f"close: PWM number is missing {pwm_num}")
+        return self.__pwm_1.close_pwm() if pwm_num == PWMPins.PWM1 else\
+               self.__pwm_2.close_pwm()
+
+    def __check_pwm_device_and_instantiate(self, pwm_num: PWMPins):
+        # Check if it is first time being called, __pwm
+        if pwm_num == PWMPins.PWM1 and self.__pwm_1 is None:
+            return PwmDevice(channel=self.__pwm_pin_to_channel[PWMPins.PWM1].value.channel,
+                                chip=self.__pwm_pin_to_channel[PWMPins.PWM1].value.chip)
+        if pwm_num == PWMPins.PWM2 and self.__pwm_2 is None:
+            return PwmDevice(channel=self.__pwm_pin_to_channel[PWMPins.PWM2].value.channel,
+                                chip=self.__pwm_pin_to_channel[PWMPins.PWM2].value.chip)
+        return None
+
+    def init_pwm(self, pwm_num: PWMPins):
+        """
+        Initializing and opening PWM device
+        Args:
+            pwm_num (PWMPins): PWM number enum
+        Return:
+            N/A
+        """
+        if pwm_num is None:
+            raise ValueError(f"init_pwm: PWM number is missing {pwm_num}")
+        pwm_dev = self.__check_pwm_device_and_instantiate(pwm_num)
+        if pwm_dev is not None:
+            pwm_dev.open_pwm()
+            self.log.debug(f"init_pwm: Instantiating PWM Device for the first time {pwm_dev.chip}"
+                           f",{pwm_dev.channel}")
+            if pwm_num == PWMPins.PWM1:
+                self.__pwm_1 = pwm_dev
+            else:
+                self.__pwm_2 = pwm_dev
+        self.log.debug("init_pwm: PWM device is already open")
```

Comment:
I think you do not need `__check_pwm_device_and_instantiate`. 
I think you can simplify everything like this:

```py
    def init_pwm(self, pwm_num: PWMPins):
        """
        Initializing and opening PWM device
        Args:
            pwm_num (PWMPins): PWM number enum
        Return:
            N/A
        """
        if pwm_num == PWMPins.PWM1 and self.__pwm_1 is None:
               self.__pwm_1 = __init_pwm_dev(pwm_num)
               return 

        if pwm_num == PWMPins.PWM2 and self.__pwm_2 is None:
               self.__pwm_2 = __init_pwm_dev(pwm_num)
               return
    
       self.log.debug("init_pwm: PWM device is already initialized and open")
```

```py
    def __init_pwm_dev(self, pwm_num: PWMPins):
            pwm_dev =  PwmDevice(channel=self.__pwm_pin_to_channel[pwm_num].value.channel,
                                chip=self.__pwm_pin_to_channel[pwm_num].value.chip)
            pwm_dev.open_pwm()
            self.log.debug(f"__init_pwm_dev: Instantiating PWM Device for the first time {pwm_dev.chip}"
                           f",{pwm_dev.channel}")
```